packages/cascara/src/ui/Table/TechnicalDesign.mdx

```diff
@@ -0,0 +1,138 @@
+
+### Triggers
+
+Custom actions are present in most tables, they are an important aspect of all applications. The nature of the actions depends on the entity they are applied to, hence, an abstract way of defining custom actions is required.
+
+| Attribute | Type     | default      | Description                                                 |
+| --------- | -------- | ------------ | ----------------------------------------------------------- |
+| id        | String   | `''`         | The id of the action, usually the name.                     |
+| callback  | Function | `noop`       | The function to be called when the action button is clicked |
+| trigger   | ANY      | `<Button />` | The element that will trigger the action                    |
+
+### Data
+
+The data passed to the table for rendering purposes.
+
+| Name | Type          | default | Description                                   |
+| ---- | ------------- | ------- | --------------------------------------------- |
+| data | Array[Object] | `[]`    | the tabular data to be displayed by the table |
+
+### Columns
+
+The definition of columns depends on the data, we must make sure we have a well-defined set of data types to allow display and mutation for each one.
+
+| Attribute  | Type                                                                   | default    | Description                                              |
+| ---------- | ---------------------------------------------------------------------- | ---------- | -------------------------------------------------------- |
+| accessor   | String                                                                 | `null`     | The name of the entity attribute that maps to the column |
+| context    | Object                                                                 | `{}`       | the context to be passed down to the custom component    |
+| isEditable | Boolean                                                                | `FALSE`    | Specifies if the cells are editable                      |
+| label      | String                                                                 | `''`       | The text to display as column header                     |
+| render     | JSX                                                                    | `null`     | A custom component to render                             |
+| style      | Object                                                                 | `{}`       | Custom style for the rendered component                  |
+| type       | Enum[`boolean`, `date`, `image`, `link`, `number`, `string`, `render`] | `'string'` | Specifies the type of data for the column                |
+
+### Data types
+
+The data types this table can handle are the following:
+
+For display puposes: `boolean`, `date`, `image`, `link`, `number`, `string` and `render`. The `render` type being a custom JSX.
```

Comment:
We should figure out how to identify which of these is part of MVP and which is future.

Also, we need to figure out what these little guys are called. They will be used in a reusable manner all over Cascara. Basically they are like display atoms or something. And they will have an editing experience as well.

- [ ] Maybe we just call these guys "atoms". And for all atoms they have an editing or display presentation?

Reply:
yes, `atoms` is a more descriptive term.

Reply:
@brianespinosa, I double-checked the apps to see which atoms are to be included in the MVP.
The lists are the following:

Display
- boolean
- date
- link
- number
- string

Edit
- switch
- input
- datetime picker
- select
- lookup select

The atoms we use for`display` will be different to the atoms we use  for `edit`, that doubles the effort.

Reply:
For "edit" the experience needs to be more simple for MVP. I think for boolean we can only do a default html checkbox. For date, we should show an input date time field, not our own date picker. Basically the edit experience for MVP is going to be: whatever is available in html and what do browser vendors provide natively. We will absolutely have other options though in the future:

- switch versus checkbox for booleans
- custom date picker to be consistent across all browsers (except I think keeping native pickers on mobile)
- custom select/dropwdown component of our own so we can control the UI/UX (again I think the exception would be mobile)
- lots of others

Reply:
Ok sounds good! Thanks for reminding me about HTML5, the mobile functionality aspect is very important.

I mentioned the switch instead of checkbox because wee have tables that implement it. At the end of the day we can style an HTML5 checkbox to make it look/behave as a switch.  I have some ideas on that matter that I want to explore and run by @CaffeinatedAllie.

Reply:
Yep. I think with these atoms, there are also use cases where we will want a checkbox too. Eventually someone could decide which control they want to show for their boolean. Does a checkbox, a switch, an enum select make the most sense?

Reply:
yes, they completely make sense. The enum select will have the `true`, `false` values - we could also allow the user to set the labels to support localizaton.

You just uncovered a track we can start to follow in terms of abstraction, looking forward to it!

Reply:
I will start typing these ideas into code soon, these kind of convos make really think about implementation! 🙌